### PR TITLE
reduce github.com/pkg/errors usage

### PR DESCRIPTION
**Pull request by pacoxu**
fix part of https://github.com/google/cadvisor/issues/3227

- [x] cadvisor directly usage  except `container/containerd/errdefs/grpc.go`
- [ ] `container/containerd/errdefs/grpc.go` is using `.cause()` and `withstack` part. Not easy to replace it with `errors` and `fmt.Errorf`
- [x] github.com/Microsoft/go-winio@v0.4.15 upgrade to v0.6.0 to remove 
- [ ] github.com/aws/aws-sdk-go@v1.35.24 github.com/pkg/errors@v0.9.1
- [ ] github.com/containerd/typeurl@v1.0.2 github.com/pkg/errors@v0.9.1
- [ ] gotest.tools/v3@v3.0.2 github.com/pkg/errors@v0.8.1
- [ ] github.com/prometheus/common@v0.32.1 github.com/pkg/errors@v0.9.1
- [ ] github.com/prometheus/common@v0.26.0 github.com/pkg/errors@v0.9.1
- [ ] github.com/prometheus/common@v0.10.0 github.com/pkg/errors@v0.8.1
- [ ] github.com/prometheus/common@v0.4.1 github.com/pkg/errors@v0.8.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	cloud.google.com/go/compute v1.7.0
-	github.com/Microsoft/go-winio v0.4.15
```
Comment (iwankgb) [inline go.mod:7]: This change seems to be breaking the build.

Comment (pacoxu) [reply go.mod:7]: > This change seems to be breaking the build.

The main change is about the use of `errors.Is`. If we do not use pkg/errors, we have to alter it to use strings.Contains when checking if an error is wrapped.

**Comment by iwankgb** [comment]
@pacoxu - e2e tests failure is a [known issue](https://kubernetes.slack.com/archives/C7J9RP96G/p1676924048501059).

**Comment by pacoxu** [comment]
> @pacoxu - e2e tests failure is a [known issue](https://kubernetes.slack.com/archives/C7J9RP96G/p1676924048501059).

This is a flake or it is always failing?

**Comment by iwankgb** [comment]
It's been falling consistently for a while. The errors are caused by unexpectedly high memory usage.

**Pull request closed without merging**